exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebServer.java

```diff
@@ -408,5 +448,47 @@ public void close() throws Exception {
     if (embeddedJetty != null) {
       embeddedJetty.stop();
     }
+    //Deleting temp directory
+    FileUtils.deleteDirectory(getTmpJavaScriptDir());
+  }
+
+  /* Static wrapper text for JavaScript content */
+  final String FILE_CONTENT_HEADER =
+      "  //Returns description\n" +
+      "  function getDescription(optionName) {\n" +
+      "    if (!(optionName in optsDescripMap)) {\n" +
+      "      return \"\";\n" +
+      "    }\n" +
+      "    return optsDescripMap[optionName];\n" +
+      "  }\n" +
+      "  var optsDescripMap = {";
+  final String FILE_CONTENT_FOOTER =
+      "};\n";
+
+  //Generate Options Description JavaScript
+  private void generateOptionsDescriptionJSFile() throws IOException {
+    //Obtain properties from Resource file
+    Properties optionsDescriptions = new Properties();
+    InputStream resrcInputStream = this.getClass().getClassLoader().getResourceAsStream(OPTIONS_DESCRIBE_PROPERTIES);
```

Comment:
This is nice, but using a ResourceBundle would be even better since that will allow the descriptions to be localized without code change.

Reply:
Let's skip localization for now :)
I'll work towards setting up the PR for DRILL-4699's PR as well, which would be a minor patch over this.